src/base.js

```diff
@@ -13,8 +13,228 @@
+};
+
+VRDisplay.prototype.requestPresent = function(layer) {
+  var self = this;
+  this.layer_ = layer;
+
+  return new Promise(function(resolve, reject) {
+    if (!self.capabilities.canPresent) {
+      reject(new Error('VRDisplay is not capable of presenting.'));
+      return;
+    }
+
+    self.waitingForPresent_ = false;
+    if (layer && layer.source) {
+      function onFullscreenChange() {
+        var fullscreenElement = Util.getFullscreenElement();
+
+        self.isPresenting = fullscreenElement === layer.source;
+        self.fireVRDisplayPresentChange_();
+        if (self.isPresenting) {
+          if (screen.orientation && screen.orientation.lock)
+            screen.orientation.lock('landscape-primary');
+          self.waitingForPresent_ = false;
+          self.beginPresent_();
+          resolve();
+        } else {
+          if (screen.orientation && screen.orientation.unlock)
+            screen.orientation.unlock();
+          self.wakelock_.release();
+          self.endPresent_();
+          self.removeFullscreenListeners_();
+        }
+      }
+      function onFullscreenError() {
+        if (!self.waitingForPresent_)
+          return;
+
+        self.removeFullscreenListeners_();
+
+        self.wakelock_.release();
+        self.waitingForPresent_ = false;
+        self.isPresenting = false;
+
+        reject(new Error('Unable to present.'));
+      }
+
+      self.addFullscreenListeners_(layer.source,
+          onFullscreenChange, onFullscreenError);
+
+      if (Util.requestFullscreen(layer.source)) {
+        self.wakelock_.request();
+        self.waitingForPresent_ = true;
+      } else if (Util.isIOS()) {
+        // *sigh* Just fake it.
+        self.wakelock_.request();
+        self.isPresenting = true;
+        self.fireVRDisplayPresentChange_();
+        self.beginPresent_();
+        resolve();
+      }
+    }
+
+    if (!self.waitingForPresent_ && !Util.isIOS()) {
+      Util.exitFullscreen();
+      reject(new Error('Unable to present.'));
+    }
+  });
+};
+
+VRDisplay.prototype.exitPresent = function() {
+  var wasPresenting = this.isPresenting;
+  var self = this;
+  this.isPresenting = false;
+  this.layer_ = null;
+  this.wakelock_.release();
+
+  return new Promise(function(resolve, reject) {
+    if (wasPresenting) {
+      if (!Util.exitFullscreen() && Util.isIOS()) {
+        self.endPresent_();
+      }
+
+      resolve();
+    } else {
+      reject(new Error('Was not presenting to VRDisplay.'));
+    }
+  });
+};
+
+// This returns an array because future versions of the spec may accept multiple
+// layers in requestPresent, and it's easier to overload function parameters
+// than it is return types.
+VRDisplay.prototype.getLayers = function() {
+  if (this.layer_) {
+    return [this.layer_];
+  }
+  return null;
+};
+
+VRDisplay.prototype.fireVRDisplayPresentChange_ = function() {
+  var event = new CustomEvent('vrdisplaypresentchange', { 'vrdisplay': this });
```

Comment:
I was going to reply here, but I instead adding a comment to the issue filed in the spec repo. Feel free to drop a line there: MozVR/webvr-spec#15.